### PR TITLE
Add report builder reports to "Reports" dropdown

### PR DESCRIPTION
**Pull request by NoahCarnahan**
Show the two most recently modified report builder reports in the "Reports" drop
down nav. The left hand side navigation remains unchanged.
@kaapstorm 

This is what it looks like:
screen shot 2016-05-30 at 3 15 02 pm

### CODE REVIEW

File: corehq/reports.py [hunk 1]
```diff
@@ -193,41 +196,78 @@ def _safely_get_report_configs(project_name):
...
+            if not config._id.startswith(CUSTOM_REPORT_PREFIX)
+            else CustomConfigurableReportDispatcher.slug
+        )
+        return reverse(slug, args=[domain, config._id])
+
+    @classmethod
+    def show_in_navigation(cls, domain=None, project=None, user=None):
+        if show_in_nav:
+            return config.visible or (
+            user and toggles.USER_CONFIGURABLE_REPORTS.enabled(user.username))
```
Comment (dimagimon) [inline corehq/reports.py]: E122 continuation line missing indentation or outdented

File: corehq/tabs/tabclasses.py [hunk 2]
```diff
@@ -53,39 +53,64 @@ def view(self):
...
+    @staticmethod
+    def _filter_sidebar_items(sidebar_items):
+        """
+        Exclude sidebar items where `item["show_in_navigation"] == False`
+        """
+        filtered_sidebar_items = []
+        for section, items in sidebar_items:
+            filtered_items = []
+            for item in items:
+                if item.has_key("show_in_navigation") and not item["show_in_navigation"]:
```
Comment (dimagimon) [inline corehq/tabs/tabclasses.py]: W601 .has_key() is deprecated, use 'in'

File: corehq/reports.py [hunk 3]
```diff
@@ -193,41 +196,78 @@ def _safely_get_report_configs(project_name):
...
+        return False
+
+    @classmethod
+    def display_in_dropdown(cls, domain=None, project=None, user=None):
+        if show_in_dropdown:
+            return config.visible or (
+            user and toggles.USER_CONFIGURABLE_REPORTS.enabled(user.username))
```
Comment (dimagimon) [inline corehq/reports.py]: E122 continuation line missing indentation or outdented

**Comment by czue** [comment]
nice!

File: corehq/reports.py [hunk 4]
```diff
@@ -193,41 +196,78 @@ def _safely_get_report_configs(project_name):
...
+        return False
+
+    return type('DynamicReport{}'.format(config._id), (GenericReportView,), {
+        'name': config.title,
+        'description': config.description or None,
+        'get_url': get_url,
+        'show_in_navigation': show_in_navigation,
+        'display_in_dropdown': display_in_dropdown,
```
Comment (kaapstorm) [inline corehq/reports.py]: Considering these are the same except for one variable, you could probably define the function just once, and use `partial`, something like so:
```python
from functools import partial

def mk_foobar_class(foo, bar):

    @classmethod
    def display_in(cls, domain=None, project=None, user=None, where=None):
        return (
            config.visible or (
                user and 
                toggles.USER_CONFIGURABLE_REPORTS.enabled(user.username)
            )
        ) if where else False

    return type('FooBar', (object,), {
        'show_in_foo': partial(display_in, where=foo),
        'display_in_bar': partial(display_in, where=bar),
    })
```

File: corehq/tabs/tabclasses.py [hunk 5]
```diff
@@ -53,39 +53,64 @@ def view(self):
...
     @property
     def sidebar_items(self):
+        tools = self._get_tools_items()
+        report_builder_nav = self._get_report_builder_items()
+        project_reports = ProjectReportDispatcher.navigation_sections(
+            request=self._request, domain=self.domain)
+        custom_reports = CustomProjectReportDispatcher.navigation_sections(
+            request=self._request, domain=self.domain)
+        sidebar_items = tools + report_builder_nav + project_reports + custom_reports
+        return self._filter_sidebar_items(sidebar_items)
```
Comment (kaapstorm) [inline corehq/tabs/tabclasses.py]: Nice refactor

File: corehq/reports.py (hunk 4 above)
Comment (NoahCarnahan) [inline corehq/reports.py]: Good point! f81dac89f6d4047ba06c4da388fa279d64a940e6

File: corehq/reports.py [hunk 6]
```diff
@@ -211,26 +211,21 @@ def get_url(cls, domain, **kwargs):
...
-            user and toggles.USER_CONFIGURABLE_REPORTS.enabled(user.username))
-        return False
+    def get_show_item_method(additional_requirement):
+        @classmethod
+        def show_item(cls, domain=None, project=None, user=None):
+            return additional_requirement and (
+                config.visible or
+                (user and toggles.USER_CONFIGURABLE_REPORTS.enabled(user.username))
+            )
+        return show_item
```
Comment (kaapstorm) [inline corehq/reports.py]: Nice!

**Comment by NoahCarnahan** [comment]
✅ ready to merge

**Pull request merged**